### PR TITLE
Inset map viz

### PR DESCRIPTION
**Pull request by elmeraa**
## Summary of changes
-  Added new targets in `2_process.R` to transform `p2_prop_western_2_StaID_metadata` by filtering to the unique staID's from `p2_prop_western_2_StaID` then converted lat/long to geometry points & included code to pull in state borders from `spData::us_states`  with additional target to filter for the 11 western states.
-  added new plotting function `plot_inset.R` that plots all state boundaries, the highlighted 11 western states, and the station and exports png with transparent background to `3_visualize/out`

### With stations 
western_states_station_inset
### Without
western_states_inset

Couple notes:
- line thickness could probably be bigger, especially depending on the dimensions of the inset (wasn't so sure on that)
- grey color can obviously be adjusted, and so can the alpha to make the state lines more visible
- the `ms_simplify()` on the state boundaries looks pretty stiff but probably won't matter as much when its a smaller inset

### CODE REVIEW

**Comment by elmeraa** [comment]
Couple notes:
- line thickness could probably be bigger, especially depending on the dimensions of the inset (wasn't so sure on that)
-  grey color can obviously be adjusted, and so can the alpha to make the state lines more visible

**Review by hcorson-dosch — CHANGES_REQUESTED** [review]
👍👍 _Really_ nice work on this Elmera, and thank you for getting this PR up so quickly. Your targets set up is neat and clean, and it ran perfectly for me locally.

After testing it in the site I've requested a few changes (comments prefaced with 'RC'). I've also made a couple suggested changes (prefaced with 'SC') that you can take or leave, and left a couple of general comments ('GC') with food for thought on pipeline structure.

**Comment by elmeraa** [comment]
## Summary of requested changes
-  moved targets from `2_process.R` to `3_visualize` - I agree, it does make more sense there
- reduced the need for excess targets that are not used downstream such as `p2_prop_western_2_StaID_metadata_sf` and  updated `p2_states_western` to reduce the need of extra code.
- added if statement to `plot_inset` function that displays stations on map when `stations =TRUE` and added additional target `p3_inset_stations_map` for seperatly outputting inset with stations displayed.
- additional aesthetic, coloring, and sizing edits

## With stations
western_states_stations_inset

## Without Stations 
western_states_inset

File: 3_visualize/src/plot_inset.R [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
+plot_inset <- function(station_data, western_us_data, us_data){
+  
+  ggplot() + 
+    geom_sf(data = us_data,
+            fill = NA,
+            color = 'black') +
+    geom_sf(data = western_us_data,
+            fill = '#DDE2E3',
```
[the remark this answers is not in the document]
Comment (elmeraa) [reply 3_visualize/src/plot_inset.R]: without `alpha = 0.5`, it ends up being a bit too dark. 
Screen Shot 2022-11-17 at 12 30 12 PM

**Review by hcorson-dosch — APPROVED** [review]
🎉 Looks great! Thanks for making those edits. I think you got the simplification on the states just right 👌

Down the road we may decide to switch all or part of the map to svg, but TBD on that.

**Pull request merged**